src/lib/db/public-signup-token-store.ts

```diff
@@ -163,14 +174,15 @@ export class PublicSignupTokenStore implements IPublicSignupTokenStore {
     }
 
     async get(key: string): Promise<PublicSignupTokenSchema> {
-        const row = await this.makeTokenUsersQuery()
-            .where('secret', key)
-            .first();
+        const rows = await this.makeTokenUsersQuery().where(
+            'tokens.secret',
+            key,
+        );
 
-        if (!row)
+        if (!rows[0])
```

Comment:
If I'm reading this right, this is actually checking that we got at least one result, right? In that case, maybe it'd be more explicit to check the array length instead of the first element?

```suggestion
        if (!rows.length)
```